homeassistant/components/maico/sensor.py

```diff
@@ -0,0 +1,161 @@
+
+#     def __init__(self, interval, maico_url):
+#         """Init Maico Data object."""
+#         self.maico_details = None
+#         self.update = Throttle(interval)(self._update)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add sensors for passed config_entry in HA."""
+    coordinator: MaicoUpdater = hass.data[DOMAIN][config_entry.entry_id]
+    created = set()
+    all_sensors = coordinator.data["sensors"]
+
+    @callback
+    def _create_entity(key: str) -> MaicoSensor:
+        """Create a sensor entity."""
+        created.add(key)
+        # data = coordinator.data["sensors"][key]
+        # description = ENTITY_DESCRIPTION_KEY_MAP.get(
+        #     data.getUnit(), IotaWattSensorEntityDescription("base_sensor")
+        # )
+        description = MaicoSensorEntityDescription(
+            "capteur maico", state_class=SensorStateClass.MEASUREMENT
+        )
+
+        return MaicoSensor(
+            coordinator=coordinator,
+            key=key,
+            entity_description=description,
+        )
+
+    async_add_entities(_create_entity(key) for key in all_sensors)
+
+    @callback
+    def new_data_received():
+        """Check for new sensors."""
+        entities = [_create_entity(key) for key in all_sensors if key not in created]
+        async_add_entities(entities)
+
+    coordinator.async_add_listener(new_data_received)
+
+
+@dataclass
+class MaicoSensorEntityDescription(SensorEntityDescription):
+    """Class describing Maico sensor entities."""
+
+    value: Callable | None = None
+
+
+class MaicoSensor(CoordinatorEntity[MaicoUpdater], SensorEntity):
+    """Defines a IoTaWatt Energy Sensor."""
+
+    entity_description: MaicoSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: MaicoUpdater,
+        key: str,
+        entity_description: MaicoSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator=coordinator)
+
+        self._key = key
+        # data = self._sensor_data
+        # if data.getType() == "Input":
+        #     self._attr_unique_id = (
+        #         f"{data.hub_mac_address}-input-{data.getChannel()}-{data.getUnit()}"
+        #     )
+        self.entity_description = entity_description
+
+    @property
+    def _sensor_data(self):
+        """Return sensor data."""
+        if self._key in self.coordinator.data["sensors"]:
+            return self.coordinator.data["sensors"][self._key]
+
+    @property
+    def name(self) -> str | None:
+        """Return name of the entity."""
+        if self._key in self.coordinator.data["sensors"]:
+            return self._key
+        return None
+
+    @property
+    def device_info(self) -> entity.DeviceInfo:
+        """Return device info."""
+        return entity.DeviceInfo(
+            manufacturer="Maico",
+            model="WS320KB",
```

Comment:
Is this supposed to be hard-coded? Would there be a way to get the model from the device's API?

Reply:
There isn't device API. but with scraping a web page, I detect a call to a cgx witch return an xml.
This xml contains VMC sensors.
There some firmware version information, but I don't know how to scrape information about model.

I don't know too if there are some other compatible models for this manufacturer.